test/modules/ModuleLib/ModuleUtil.cs

```diff
@@ -57,11 +58,29 @@ public static ILogger CreateLogger(string categoryName, LogEventLevel logEventLe
         }
 
         public static async Task ReportTestResultAsync(TestResultReportingClient apiClient, ILogger logger, TestResultBase testResult)
-        {            
+        {
             logger.LogInformation($"Sending test result: Source={testResult.Source}, Type={testResult.ResultType}, CreatedAt={testResult.CreatedAt}, Result={testResult.GetFormattedResult()}");
             await apiClient.ReportResultAsync(testResult.ToTestOperationResultDto());
         }
 
+        public static async Task ReportTestResultUntilSuccessAsync(TestResultReportingClient apiClient, ILogger logger, TestResultBase testResult, CancellationToken cancellationToken)
+        {
+            bool isSuccessful = false;
+            while(!isSuccessful && !cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    logger.LogInformation($"Sending test result: Source={testResult.Source}, Type={testResult.ResultType}, CreatedAt={testResult.CreatedAt}, Result={testResult.GetFormattedResult()}");
```

Comment:
you should call ReportTestResultAsync here

Reply:
why do you need to retry to send result to TRC?

Reply:
This is more of a development function that I'm using to handle the TRC exception that it thrown b/c the sources are invalid. I'll remove portion of code to be used `ReportTestResultAsync` once the TRC has a proper source. 

https://msazure.visualstudio.com/One/_workitems/edit/6186177